Fix issue #15597 — Mention dropup variation in dropdown docs

kkirsche opened the pull request:
Fixes #15597. Mention dropup variation in dropdown docs

Other pull request (#15659) was so messed up I sincerely have no clue how to save it. The recommended command doesn't work, can't believe rebasing is so problematic...

kkirsche committed:
Fix issue #15597 — Mention dropup variation in dropdown docs

Fix issue #15597 — Mention dropup variation in dropdown docs

Other pull request was so messed up I sincerely have no clue how to save it. The recommended command doesn't work, can't believe rebasing is so problematic...

cvrebert commented:
CC: @mdo

vsn4ik commented:
- Remove `<h3>Example</h3>`;
- Move `data-example-id=""` to `<h3>Dropup Variation</h3>` and change value;
- We must somehow keep the style of `.dropdown` (`position: static`), but I do not know how to do it.

kkirsche commented:
I'm not sure I follow you @vsn4ik on point 3

Commit:
Complete @vsn4ik's requests

* Remove `<h3>Example</h3>;`
* Move `data-example-id=""` to `<h3>Dropup Variation</h3>` and change value;

vsn4ik commented:
My mistake on point 2: `data-example-id` not need be moved. Only change, because `data-example-id="static-dropdown"` already exists.

Commit:
Update per @vsn4ik's requests

data-example-id not need be moved. Only change, because data-example-id="static-dropdown" already exists.

Remove whitespace after h3.
Variation not need capitalize.

mdo commented on docs/_includes/components/dropdowns.html:
```diff
@@ -33,6 +33,36 @@ <h3 id="dropdowns-example">Example</h3>
+  <h3 id="dropdowns-dropup">Dropup variation</h3>
+  <p>Trigger dropdown menus above elements by adding <code>.dropup</code> to the parent.</p>
+  <div class="bs-example" data-example-id="static-dropup">
+    <div class="dropup clearfix">
```
Did I add the `clearfix` at some point, or is that new to this PR?

mdo commented:
My only concern with this is that it's basically an exact dupe of another docs section.

kkirsche commented:
I didn't add one to this. I copied and pasted to use what you had. 

I had started with a small note only, what would you be interested in as a solution?

mdo commented:
Yeah, I think so @kkirsche.

kkirsche commented:
@mdo What do you think about the following idea:

In the example section we add the following sentence:

```
Dropdown menus can also be triggered above elements by adding <code>.dropup</code> to the parent.
```

Then in the example section include one of each. This way, we reduce the space necessary by not including a standalone example for just the dropup variation.

The other idea I have would be to include the following sentence with no visible example:
```
Dropdown menus can also be triggered above elements, refer to <a href="components/#btn-dropdowns-dropup">Button group dropup variation</a> for more information and examples.
```

What is your preference?

mdo commented:
A live example to go with it is probably preferable, so we can go with that.

Commit:
Remove separate area for dropup's

Condense the dropup example

kkirsche commented:
Changed to the first option which uses short sentence with a live example in the first example box.

cvrebert commented:
@mdo LGTY?

mdo commented:
:shipit:

kkirsche commented:
Thanks guys! :+1:
